pytorch_lightning/utilities/imports.py

```diff
@@ -85,6 +85,25 @@ def _compare_version(package: str, op: Callable, version: str, use_base_version:
     return op(pkg_version, Version(version))
 
 
+class _RequirementAvaliable:
```

Comment:
Typo:

```suggestion
class _RequirementAvailable:
```

Reply:
it is a replacement of `_compare_version` and what is the advantage of class over function?

Reply:
This is much more than `_compare_version`. It checks an entire dependency, including the package, install+extras requires and version specifiers. Could be used to replace pairs of `_package_available` and `_compare_version` or even `_module_available` if in some case it is implied by an extra require.

The benefit of having a class is that it is not just a boolean. It also gives a clear message saying why the dependency is not met. This is being used in `LightningCLI` to show to the user.